File: ts/src/parkingManager/radomParkingManagerService.ts
/** @format */

import {ParkingManagerService} from './parkingManagerService'
import {ParkingBoy} from '../parkingBoy/parkingBoy'

export class RandomParkingManagerService implements ParkingManagerService {
    findOneValidParkingBoy(parkingBoys: ParkingBoy[]): ParkingBoy | null {
        return (
            parkingBoys
                .filter(({hasAvailableSpaces}) => hasAvailableSpaces)
                .sort(() => Math.floor(Math.random() * parkingBoys.length))[0] || null
        )
    }
}

File: kt/src/main/kotlin/com/github/scglwsj/parking/manager/RadomParkingManagerService.kt
package com.github.scglwsj.parking.manager

import com.github.scglwsj.parking.boy.BaseParkingBoy

class RandomParkingManagerService : ParkingManagerService {
    override fun findOneValidParkingBoy(parkingBoys: List<BaseParkingBoy>): BaseParkingBoy? =
        parkingBoys.filter{ it.hasAvailableSpaces }.minBy { (0..parkingBoys.size).random() }
}
